src/BizHawk.Client.EmuHawk/CoreDownloaderForm.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+using BizHawk.Common;
+using BizHawk.Common.PathExtensions;
+
+namespace BizHawk.Client.EmuHawk
+{
+	public partial class CoreDownloaderForm : Form
+	{
+		// TODO edit
+		private static readonly string BaseUrl = $"https://github.com/Morilli/BizHawk-extra/raw/{VersionInfo.MainVersion}/";
```

Comment:
Having FFmpeg in https://github.com/TASEmulators/ffmpeg-binaries makes sense because it doesn't change. Before hosting any core binaries in a GitHub repo like that, we need to plan ahead and decide on a versioning scheme, a way to check compatibility (in case it's downloaded manually), and an expiration date for URIs (#2235). We also need to decide how dev builds should behave.

Reply:
Right, this is one of the things that need to be decided before this can work. In this implementation I was thinking of having a separate repository (just like ffmpeg-binaries) that hosts downloadable dlls for each released bizhawk version in a separate branch, so for release `2.9.2` you would have a `2.9.2` branch with the binaries from that release. Those would of course stay static after the release.

It would probably also be possible to fetch the binaries from the BizHawk repo itself using the release tag, but that feels a little more volatile in regards to future-proofness.
I see your point in that potentially breaking this download in the future because we don't control the url anymore would be bad, but I don't see how this case here differs from the ffmpeg download case.
Compatibility checking could be implemented but I personally wouldn't bother, because as long as we serve proper downloads compatibility should be guaranteed, and if people download stuff on their own, well, that's their own problem (you can already replace any non-bizhawk dlls and cores with different versions).

> We also need to decide how dev builds should behave.

I say ignore that case completely, dev builds can stay being full-size downloads and one dev `2.9.2` has little meaning when it could span an entire year of changes, so those downloads don't need to work (or be valid working files) for dev builds.